cmd/strace.go

```diff
@@ -76,22 +77,96 @@ func strace(c *cli.Context) error {
 			return nil
 		}
 
-		// Print register value on every second trap so that we print only after the syscall is complete
+		// Print register values on every second trap so that we print only after the syscall is complete
 		if traps > 0 && traps%2 == 0 {
-			origRax, err := readOrigRax(childPid)
+			var regs syscall.PtraceRegs
+			err = syscall.PtraceGetRegs(childPid, &regs)
 			if err != nil {
 				return fmt.Errorf("error while reading Orig_rax for child pid %d: %w ", childPid, err)
 			}
-			log.WithFields(log.Fields{
-				"orig_rax": origRax,
-				"syscall":  syscallNameByRax[origRax],
-			}).Info("Caught syscall")
+
+			// Syscall-related data is stored in a number of registers, namely:
+			// - rax (system call number)
+			// - rdi (first arg)
+			// - rsi (second arg)
+			// - rdx (third arg)
+			// - r10 (fourth arg)
+			// - r8 (fifth arg)
+			// - r9 (sixth arg)
+			// For each syscall, we need to hardcode which arguments are of interest
+			// Hence, for educational purposes, only selected syscalls have been
+			// defined (i.e. are recognized) below. The list can grow over time.
+			syscallName := syscallNameByRax[regs.Orig_rax]
+
+			switch syscallName {
```

Comment:
```suggestion
			switch syscallName := syscallNameByRax[regs.Orig_rax] {
```

Reply:
Also, maybe this should be a separate function? Maybe something like `printSyscall()`?

Reply:
Sure, good point.